lib/hackerone/client/report.rb

```diff
@@ -75,6 +75,32 @@ def activities
         end
       end
 
+      def program
+        @program || Program.find(relationships[:program][:data][:attributes][:handle])
+      end
+
+      def assign_to_user(name)
+        fail(
```

Comment:
Client-side validation like this is a bit tricky. It's a matter of time the API changes its validation, and this client becomes out of sync. Consumers will have a hard time figuring out what's wrong since they have multi places to check what's going on.

I would expect this client to just do the API call and use the errors from the response to generate the expectation. Having said that, I'm not exactly sure how the HackerOne API error response looks like and if you get an error that's helpful.